### PR TITLE
Fix #661: Open in Brave now opens the first URL found in share content

### LINKED ISSUE #661: Can't open Brave through share tool from feedly
### Description: 
image from ios
Nothing happened after I click Brave

FWIW, I share the article to Note and I notice it is not pure URL
It contains title like this
`
Facebook doesn't need to sell your data. It has been giving it away free for years. https://www.recode.net/2018/12/19/18148517/facebook-data-partnerships-mark-zuckerberg-new-york-times-privacy
` which is probably the root cause

### Steps to Reproduce 
  1. Share article on feedly
  2. Click Brave

**Actual result:** 

**Expected result:**

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
100%

**Brave Version:** 
1.7.1(18.12.19.01)

**Device details:** 
iPhoneX 12.1.2

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)?

### PR DESCRIPTION
**Pull request by kylehickinson**
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-ios/issues) for my issue if one did not already exist.
- [x] My patch or PR title has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!` (or `No Bug: <message>` if no relevant ticket)
- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New files have MPL-2.0 license header.

## Test Plan:

- Test that regular shared links work (Safari > Share button > Open In Brave)
- Test that sharing non-URL's fails silently (Safari > Select text on page > Share… > Open In Brave). In the future this can be used to _search_ for said text instead of just open a new tab
- Test that opening text that _contains_ a link in it works (example apps that share both title & URL: Feedly, Reeder)

## Reviewer Checklist:

- [x] PR is linked to an issue via [Zenhub](https://www.zenhub.com/extension).
- [x] Issues are assigned to at least one epic.
- [x] Issues include necessary QA labels:
  - [x] `QA/(Yes|No)`
  - [x] `release-notes/(include|exclude)`
  - [x] `bug` / `enhancement`
- [x] Necessary security reviews have taken place.
- [x] Adequate test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable)

### CODE REVIEW

**Review by iccub — APPROVED** [review]

**Comment by jhreis** [comment]
@tomlowenthal should review this before merging, per comments on the fixed ticket.

**Comment by tildelowengrimm** [comment]
@jhreis Looks like it implements the plan in #661 to me?

**Review by danishjafri88 — APPROVED** [review]

**Comment by kylehickinson** [comment]
@tomlowenthal @jhreis Does this have security team's blessing then? :)

**Comment by jhreis** [comment]
@kylehickinson can we have some unit tests added around this. At the very least testing the parsing logic?

**Review by jhreis — CHANGES_REQUESTED** [review]
Can we have some unit tests added around this. At the very least testing the parsing logic?

**Comment by kylehickinson** [comment]
The actual parsing comes from `NSDataDetector` but maybe I can move some of this logic into BraveShared and add a unit test for at least the `firstURL(in:)` method

**Comment by kylehickinson** [comment]
Random project changes are due to this warning I found in CI:

```
The file reference for "WindowRenderHelper.js" is a member of multiple groups ("AtDocumentStart" and "UserScripts"); this indicates a malformed project.  Only the membership in one of the groups will be preserved (but membership in targets will be unaffected).  If you want a reference to the same file in more than one group, please add another reference to the same path.
```

So I think adding a library ref refreshed the project and forced these updates… Decided to commit to silence them

**Comment by tildelowengrimm** [comment]
I'm not qualified to review Swift code. The behavior described in #661 is approved. I think this swift code does that, but ultimately it's the code reviewer's job to be sure of that.

**Review by iccub — APPROVED** [review]

**Review by danishjafri88 — APPROVED** [review]

File: BraveShareTo/ShareToBraveViewController.swift [hunk 1]
```diff
@@ -33,19 +34,22 @@ class ShareToBraveViewController: SLComposeServiceViewController {
...
-            var urlItem: URL!
+            var urlItem: URL?
             
             // We can get urls from other apps as a kUTTypeText type, for example from Apple's mail.app.
             if let text = item as? String {
-                urlItem = URL(string: text)
+                urlItem = text.firstURL
             } else if let url = item as? URL {
-                urlItem = url
+                urlItem = url.absoluteString.firstURL
```
Comment (jhreis) [inline BraveShareTo/ShareToBraveViewController.swift:43]: Is this necessary? If the item is a URL, it by nature, is a single URL, right?

**Review by jhreis — APPROVED** [review]

File: BraveShareTo/ShareToBraveViewController.swift (hunk 1 above)
Comment (kylehickinson) [reply BraveShareTo/ShareToBraveViewController.swift:43]: It's not super necessary, more like a just-in-case to keep logic consistent

**Pull request merged**